server/src/main/java/org/opensearch/indices/replication/SegmentReplicationState.java

```diff
@@ -25,9 +25,11 @@ public class SegmentReplicationState implements ReplicationState {
      * @opensearch.internal
      */
     public enum Stage {
-        DONE((byte) 0),
+        INIT((byte) 1),
```

Comment:
having DONE and INIT as 0 and 1 was intentional actually so new states can be added in a backward compatible way, see [here](https://github.com/opensearch-project/OpenSearch/pull/3439#discussion_r882018899).  You should only need to add the new state `REPLICATING((byte) 3)` at the end.